### PR TITLE
fix: add correct pagination to event api

### LINKED ISSUE #13508: API don't match with Documentation
## Bug description

I try to get events with API and use this documentation article (https://posthog.com/docs/api/events). Reply from API don't match with Documentation and seems broken.

Documentation for `List all events`
image

## How to reproduce

1. Call method `https://app.posthog.com/api/projects/<project id>/events`

## Actual result

image

Server didn't return `previous` and `count` properties. Link for next page don't contain `limit` and `offset` params (it use `before` for pagination)

## Expected result

Server return `previous` and `count` properties. And `next` property is correct (use `limit` and `offset` for pagination).

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: please provide

### PR DESCRIPTION
**Pull request by kappa90**
## Problem
Closes https://github.com/PostHog/posthog/issues/13508

We don't currently support in the LIST events API endpoint a `previous` parameter for correct pagination, or a `count`.

## Changes
- Adds `previous` and `count` to LIST events API result

## How did you test this code?
- Improved existing tests

### CODE REVIEW

**Comment by kappa90** [comment]
@mariusandra not sure why but one hogQL production test didn't match the API, see where I removed `hasMore` from the test.

File: posthog/api/event.py [hunk 1]
```diff
@@ -159,22 +190,35 @@ def list(self, request: request.Request, *args: Any, **kwargs: Any) -> response.
...
+                    previous_url = self._build_pagination_url(
+                        request, "previous", query_result[0]["timestamp"], order_by
+                    )
+
+            return response.Response(
+                {
+                    "next": next_url,
+                    "previous": previous_url,
+                    "results": result,
+                    "count": len(result),
```
Comment (timgl) [inline posthog/api/event.py:219]: This'll be wrong as the result is limited. We also in general do not want to return a count for events as that'd mean getting _all_ events to calculate just this property.

Comment (hit2hat) [reply posthog/api/event.py:219]: Hello @timgl!

Maybe you make separated method for getting count of events? It's important thing for some cases.

For example: we use PostHog as a main analytics tool and would like to make Results of Year for our users, but without `count` property we can't calculate events quickly and have big problems.

Comment (neilkakkar) [reply posthog/api/event.py:219]: Hey @hit2hat the separate method you want to use here to get total counts is the `trends` API, which does this aggregation over a time range. (it's the same as what you see for the graphs on the trends insight inside PostHog)

Comment (hit2hat) [reply posthog/api/event.py:219]: Hello!

Seems like a solution, thanks. But it's may be not clear for other PostHog users.

Comment (kappa90) [reply posthog/api/event.py:219]: @timgl maybe I misunderstand how querying clickhouse works, but isn't the count I'm returning here just the length of the limited query result (see line 196-198), which we are returning anyways?

Comment (neilkakkar) [reply posthog/api/event.py:219]: Yes, @kappa90 , but as you say, it's not very useful if it's only the limited count (you already can compute this data) & not the total count of events in the time range.

In such an API, the count is expected to return the total, not the count after applying the limits.

**Pull request closed without merging**